### PR TITLE
feature(cluster_k8s): add autoscaling feature

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/qcZmndR3/2690-gke-add-autoscale-feature

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by dkropachev** [comment]
Turned out that autoscaling won't work due to the local ssd limitation:
https://cloud.google.com/kubernetes-engine/docs/concepts/cluster-autoscaler#limitations
We need to schedule nodes manually

File: sdcm/sct_config.py [hunk 1]
```diff
@@ -675,6 +676,17 @@ class SCTConfiguration(dict):
...
+
+        dict(name="k8s_min_scylla_nodes", env="SCT_K8S_MIN_SCYLLA_NODES", type=str,
+             help="Minimum number of nodes in the pool. "
+                  "Autoscaler stop draining k8s nodes if scylla pods getting lower this number"),
+
+        dict(name="k8s_max_scylla_nodes", env="SCT_K8S_MAX_SCYLLA_NODES", type=str,
+             help="Maximum number of nodes in the pool. "
+                  "Autoscaler stop creating k8s nodes after scylla pods getting higher than this number."
+                  "When you hit this number, pod creation is getting failed"),
+
```
Comment (enaydanov) [inline sdcm/sct_config.py:689]: You need to update doc too.

File: sdcm/cluster_k8s/gke.py [hunk 2]
```diff
@@ -107,6 +112,14 @@ def setup_gke_cluster(self, num_nodes: int) -> None:
         LOGGER.info("Create GKE cluster `%s' with %d node(s) in default-pool and 1 node in operator-pool",
                     self.gke_cluster_name, num_nodes)
         tags = ",".join(f"{key}={value}" for key, value in self.tags.items())
+
+        if self.min_nodes_num is not None or self.max_nodes_num is not None:
+            autoscale_option = f" --enable-autoscaling"
+            autoscale_option += '' if self.min_nodes_num in [None, ''] else f' --min-nodes {self.min_nodes_num}'
+            autoscale_option += '' if self.max_nodes_num in [None, ''] else f' --max-nodes {self.max_nodes_num}'
```
Comment (enaydanov) [inline sdcm/cluster_k8s/gke.py:119]: `if self.min_nodes_num` and `if self.max_nodes_num` is enough here.

File: sdcm/cluster_k8s/gke.py [hunk 3]
```diff
@@ -107,6 +112,14 @@ def setup_gke_cluster(self, num_nodes: int) -> None:
         LOGGER.info("Create GKE cluster `%s' with %d node(s) in default-pool and 1 node in operator-pool",
                     self.gke_cluster_name, num_nodes)
         tags = ",".join(f"{key}={value}" for key, value in self.tags.items())
+
+        if self.min_nodes_num is not None or self.max_nodes_num is not None:
+            autoscale_option = f" --enable-autoscaling"
```
Comment (enaydanov) [inline sdcm/cluster_k8s/gke.py:117]: `f` is not required here

File: sdcm/cluster_k8s/gke.py [hunk 4]
```diff
@@ -121,8 +134,9 @@ def setup_gke_cluster(self, num_nodes: int) -> None:
                        f" --local-ssd-count {self.gce_n_local_ssd}"
                        f" --node-taints role=scylla-clusters:NoSchedule"
                        f" --enable-stackdriver-kubernetes"
+                       f"{autoscale_option}"
                        f" --no-enable-autoupgrade"
-                       f" --no-enable-autorepair"
+                       f" --enable-autorepair"
```
Comment (enaydanov) [inline sdcm/cluster_k8s/gke.py:139]: Do we want to change this parameter in Scylla Operator deployment guide too? I think it's better to stay with `--no-enable-autorepair` for jobs which do not autoscale.

I.e.:

```python
f" {'--enable-autorepair' if autoscale_option else '--no-enable-autorepair'}"
```

File: test-cases/scylla-operator/longevity-scylla-operator-3h-gke.yaml [hunk 5]
```diff
@@ -6,7 +6,7 @@ n_db_nodes: 3
 n_loaders: 2
 n_monitor_nodes: 1
 
-nemesis_class_name: 'KubernetesScyllaOperatorMonkey'
+nemesis_class_name: 'AddRemoveRackNemesis'
```
Comment (enaydanov) [inline test-cases/scylla-operator/longevity-scylla-operator-3h-gke.yaml:9]: Leftover?

**Pull request closed without merging**